src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +140,103 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
```

Comment:
"Producer" isn't really the most appropriate name.
I'd expect a "producer" to be like a factory which can create more and more items.
"TaskCompletionSource" isn't too niche a thing in .NET. They're often just named "tcs" in .NET source code
```suggestion
		var sessionIdTcs = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
```